mono/mini/method-to-ir.c

```diff
@@ -3976,7 +3976,7 @@ mini_emit_ldelema_1_ins (MonoCompile *cfg, MonoClass *klass, MonoInst *arr, Mono
 		static const int fast_log2 [] = { 1, 0, 1, -1, 2, -1, -1, -1, 3 };
 
 		EMIT_NEW_X86_LEA (cfg, ins, array_reg, index2_reg, fast_log2 [size], MONO_STRUCT_OFFSET (MonoArray, vector));
-		ins->klass = m_class_get_element_class (klass);
+		ins->klass = klass;
```

Comment:
How precise does the class have to be in IR?  `klass` here is some array type, right?  So this should be a reference (or maybe ptr) to the array element type - or would that end up generating worse code down the line?

Reply:
Its mostly used for verification.